frontend/src/components/SignUp/signup.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { Link } from "react-router-dom";
+import styles from'./signup.module.css';
+import gmail_icon from '../Logos/gmail_icon.png';
+import fb_icon from '../Logos/fb_icon.png';
+import twtx_icon from '../Logos/twtx_icon.webp';
+import apple_icon from '../Logos/apple_icon.png';
+
+function EmailErrorMessage({username, isValid}) {
+    return <p className={styles.error_message}>Please enter a valid email.</p>
+}
+
+function PasswordErrorMessage({password, isValid}) {
+    return <p className={styles.error_message}>Passwords must be at least 8 characters and 
+    include at least one lowercase, uppercase, number, and special symbol (~|!@#$%^&*).</p>
+}
+
+function PasswordConfirmation({password, isValid}) {
+    return <p className={styles.error_message}>Passwords do not match. </p>
+}
```

Comment:
No change needed -- this code will probably change once we do proper server authentication but, 
A couple nits here:

1. in general we strive to do 1 component per file do these should be in its own file
2. We could create a standardize "error message" component if we wanted to. Right now these are all the same basically, it takes it a boolean, the other argument isn't even being used.